src/display/annotation_layer.js

```diff
@@ -1832,157 +1847,237 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
-    this.container.style.top = `${(100 * (popupTop - pageY)) / pageHeight}%`;
-
-    this.container.append(popup.render());
     return this.container;
   }
 }
 
 class PopupElement {
-  constructor(parameters) {
-    this.container = parameters.container;
-    this.trigger = parameters.trigger;
-    this.color = parameters.color;
-    this.titleObj = parameters.titleObj;
-    this.modificationDate = parameters.modificationDate;
-    this.contentsObj = parameters.contentsObj;
-    this.richText = parameters.richText;
-    this.hideWrapper = parameters.hideWrapper || false;
+  #dateTimePromise = null;
 
-    this.pinned = false;
-  }
+  #boundHide = this.#hide.bind(this);
 
-  render() {
-    const BACKGROUND_ENLIGHT = 0.7;
+  #boundShow = this.#show.bind(this);
 
-    const wrapper = document.createElement("div");
-    wrapper.classList.add("popupWrapper");
+  #boundToggle = this.#toggle.bind(this);
 
-    // For Popup annotations we hide the entire section because it contains
-    // only the popup. However, for Text annotations without a separate Popup
-    // annotation, we cannot hide the entire container as the image would
-    // disappear too. In that special case, hiding the wrapper suffices.
-    this.hideElement = this.hideWrapper ? wrapper : this.container;
-    this.hideElement.hidden = true;
+  #color = null;
 
-    const popup = document.createElement("div");
-    popup.classList.add("popup");
+  #container = null;
 
-    const color = this.color;
-    if (color) {
-      // Enlighten the color.
-      const r = BACKGROUND_ENLIGHT * (255 - color[0]) + color[0];
-      const g = BACKGROUND_ENLIGHT * (255 - color[1]) + color[1];
-      const b = BACKGROUND_ENLIGHT * (255 - color[2]) + color[2];
-      popup.style.backgroundColor = Util.makeHexColor(r | 0, g | 0, b | 0);
+  #contentsObj = null;
+
+  #elements = null;
+
+  #parent = null;
+
+  #parentRect = null;
+
+  #pinned = false;
+
+  #popup = null;
+
+  #rect = null;
+
+  #richText = null;
+
+  #titleObj = null;
+
+  constructor({
+    container,
+    color,
+    elements,
+    titleObj,
+    modificationDate,
+    contentsObj,
+    richText,
+    parent,
+    rect,
+    parentRect,
+    open,
+  }) {
+    this.#container = container;
+    this.#titleObj = titleObj;
+    this.#contentsObj = contentsObj;
+    this.#richText = richText;
+    this.#parent = parent;
+    this.#color = color;
+    this.#rect = rect;
+    this.#parentRect = parentRect;
+    this.#elements = elements;
+
+    const dateObject = PDFDateString.toDateObject(modificationDate);
+    if (dateObject) {
+      // The modification date is shown in the popup instead of the creation
+      // date if it is available and can be parsed correctly, which is
+      // consistent with other viewers such as Adobe Acrobat.
+      this.#dateTimePromise = parent.l10n.get("annotation_date_string", {
+        date: dateObject.toLocaleDateString(),
+        time: dateObject.toLocaleTimeString(),
+      });
```

Comment:
Let's improve behaviour if the AnnotationLayer is used *outside* of the default viewer, where proper l10n-data may not be available.
```suggestion
      this.#dateTimePromise = parent.l10n.get(
        "annotation_date_string",
        {
          date: dateObject.toLocaleDateString(),
          time: dateObject.toLocaleTimeString(),
        },
        "{{date}}, {{time}}"
      );
```